app/models/score.rb

```diff
@@ -22,4 +25,15 @@ def serialize
   def future_score
     errors.add(:played_at, 'must not be in the future') if played_at > Time.zone.today
   end
+
+  def number_holes
+    errors.add(:number_of_holes, 'must be only 9 or 18') if
+      number_of_holes != 9 && number_of_holes != 18
+  end
+
+  def score_based_holes
+    errors.add(:number_of_holes, 'wrong range of scores based on number of holes') if
+      (total_score < 54 && total_score > 180 && number_of_holes == 18) ||
+      (total_score < 27 && total_score > 90 && number_of_holes == 9)
```

Comment:
I think these conditions are not right because, for instance, a number can't be both smaller than 54 and bigger than 180 at the same time.